Source/WebCore/loader/HistoryController.h

```diff
@@ -86,8 +85,8 @@ class HistoryController {
     RefPtr<HistoryItem> protectedProvisionalItem() const;
     void setProvisionalItem(HistoryItem*);
 
-    void pushState(RefPtr<SerializedScriptValue>&&, const String& title, const String& url);
-    void replaceState(RefPtr<SerializedScriptValue>&&, const String& title, const String& url);
+    void pushState(RefPtr<SerializedScriptValue>&&, const String&, const String& url);
+    void replaceState(RefPtr<SerializedScriptValue>&&, const String&, const String& url);
```

Comment:
> I think we can remove the parameters here instead of leaving them unused, right? They only have to remain in the History object itself?

Given that they’re still required parameters in the web-exposed JS API — per the IDL and the method signatures at https://html.spec.whatwg.org/#the-history-interface:dom-history-pushstate — then, unless there’s some special way to handle it in the generated bindings code, don’t we still need to include the parameters in the C++ sources?

And just to be clear about the way they’re used and [documented](https://developer.mozilla.org/en-US/docs/Web/API/History/pushState#javascript for example) for frontend code: developers normally just pass in the empty string as the 2nd arg:

```js
const state = { page_id: 1, user_id: 5 };
const url = "hello-world.html";
history.pushState(state, "", url);
```
That is, the JS calls still require that arg to be given (though giving it the empty string is just the normal convention for doing the minimal thing required; it could of course be any string — `"foo"` or whatever — since implementations aren’t required to actually do anything with it).

Reply:
> Given that they’re still required parameters in the web-exposed JS API — per the IDL and the method signatures at https://html.spec.whatwg.org/#the-history-interface:dom-history-pushstate — then, unless there’s some special way to handle it in the generated bindings code, don’t we still need to include the parameters in the C++ sources?

In the History object, absolutely yes. But that's not what I'm commenting on here.

There's probably no reason for HistoryController to keep these parameters, and for the History object to pass empty args along to HistoryController, right? Let's remove them wherever possible.

Reply:
> There's probably no reason for HistoryController to keep these parameters, and for the History object to pass empty args along to HistoryController, right? Let's remove them wherever possible.

I see now — yes, of course. (Thanks for your patience, and sorry for my confusion. I guess I had it stuck in my mind that the diff we were looking at there was for the History object  — despite it being clear from your review comment that this was about the HistoryController code…)